mindsdb_server/namespaces/datasource.py

```diff
@@ -66,11 +67,108 @@ def save_datasource_metadata(ds):
         with open(os.path.join(mindsdb.CONFIG.MINDSDB_DATASOURCES_PATH, ds['name'], 'metadata.json'), 'w') as fp:
             json.dump(ds, fp)
 
-def create_sqlite_db(path, ds):
+def create_sqlite_db(path, data_frame):
     con = sqlite3.connect(path)
-    ds.to_sql(name='data', con=con, index=False)
+    data_frame.to_sql(name='data', con=con, index=False)
     con.close()
 
+def get_analysis(source):
+    global global_mdb
+    return global_mdb.analyse_dataset(source, sample_margin_of_error=0.025)
+
+def cast_df_columns_types(df):
+    types_map = {
+        'Numeric': {
+            'Int': 'int64',
+            'Float': 'float64',
+            'Binary': 'bool'
+        },
+        'Date': {
+            'Date': 'datetime64',       # YYYY-MM-DD
+            'Timestamp': 'datetime64'   # YYYY-MM-DD hh:mm:ss or 1852362464
+        },
+        'Categorical': {
+            'Binary Category': 'category',
+            'Category': 'category'
+        },
+        'File Path': {
+            'Image': 'object',
+            'Video': 'object',
+            'Audio': 'object'
+        },
+        'Url': {},
+        'Sequential': {
+            'Text': 'object',
+            'Array': 'object'
+        }
+    }
+
+    analysis = get_analysis(df)
+    columns = [dict(name=x) for x in list(df.keys())]
+
+    for column in columns:
+        try:
+            name = column['name']
+            col_type = analysis['data_analysis_v2'][name]['typing']['data_type']
+            col_subtype = analysis['data_analysis_v2'][name]['typing']['data_subtype']
+            new_type = types_map[col_type][col_subtype]
+            if new_type == 'int64' or new_type == 'float64':
```

Comment:
Since some data cleaning is required here I assume more data cleaning will be required for other datasets... Mindsdb assumes the data types for the cleaned-up data.

I guess we could always have a mindsdb.clean() functionality that we run on datasource to do preliminary cleaning up but I don't know. For now, replacing `,` with `.` should cover over half the cases we are likely to stumble in, so that's fine, let's just keep this in mind.